framework/src/org/checkerframework/common/value/ValueCheckerUtils.java

```diff
@@ -149,31 +148,35 @@ public static Range getRangeFromValues(List<? extends Number> values) {
         if (range.isNothing()) {
             return values;
         }
-        if (expectedType == Integer.class
-                || expectedType == int.class
-                || expectedType == Long.class
-                || expectedType == long.class
-                || expectedType == Short.class
-                || expectedType == short.class
-                || expectedType == Byte.class
-                || expectedType == byte.class) {
-            for (Long value = range.from; value <= range.to; value++) {
-                values.add(expectedType.cast(value.longValue()));
-            }
-        } else if (expectedType == Double.class
-                || expectedType == double.class
-                || expectedType == Float.class
-                || expectedType == float.class) {
-            for (Long value = range.from; value <= range.to; value++) {
-                values.add(expectedType.cast(value.doubleValue()));
-            }
-        } else if (expectedType == Character.class || expectedType == char.class) {
-            for (Long value = range.from; value <= range.to; value++) {
-                values.add(expectedType.cast((char) value.intValue()));
+
+        // Does not overflow, because the width has already been checked.
+        long length = range.to - range.from;
+
+        // Each value is computed as a sum of the first value and an offset within the range,
+        // to avoid comparing to range.to, which may be Long.MAX_VALUE
+        for (long offset = 0; offset <= length; offset++) {
+            long value = range.from + offset;
```

Comment:
I don't see how this rewrite is any different than the logic before and this new code has an additional addition.
If `range.from` is `0` and `range.to` is `Long.MAX_VALUE`, both versions iterate over `Long.MAX_VALUE` elements. What is the benefit of using this offset?